docs/arcadia-users-group/20230228-intro-to-python-1/lesson.md

```diff
@@ -0,0 +1,548 @@
+
+You can tell if a cell has successfully executed based on the `[ ]` text to the left of the cell. A cell that is in progress has an asterisk `[*]`, and a cell that has completed has a number `[1]`.
+
+For more on Jupyter notebooks, you can check out this [AUG Lesson](https://training.arcadiascience.com/arcadia-users-group/20221024-jupyter-notebooks/lesson/). 
+
+> **Fun fact!**
+>
+> The Python programming language is not named for snakes, but instead for Monty Python, a signifier of the idea that the language should be [fun to use](https://en.wikipedia.org/wiki/Python_(programming_language)#Design_philosophy_and_features).  
+> Despite this, many Python tools and packages make reference to snakes, such as packages you may have heard us talk about before: [Anaconda](https://www.anaconda.com/products/distribution), [conda](https://docs.conda.io/en/latest/), [mamba](https://mamba.readthedocs.io/en/latest/), [snakemake](https://snakemake.readthedocs.io/en/stable/), etc.
+
+---
+## 1. Variables
+
+Variables in programming languages are containers for assigning values.
+
+Below, we assign the value `1` to the variable `x` and the value `4` to the variable `y`.
+
+
+```python
+x = 1
+y = 4
+```
+
+When we ask Python to evalute `x + y`, it substitutes the value `1` for `x` and the value `4` for `y`.
+
+Running the cell below returns the output value of the expression `x + y`.
+
+
+```python
+x + y
+```
+
+
+
+
+    5
+
+
+
+---
+## 2. Data types
+
+Variables in Python come in a variety of types, such as:
+- `int` or **integer**: whole numbers such as `1`, `2`, `3`, etc.
+- `float` or **floating-point number**: numbers including decimals, e.g. `1.03`.
+- `str` or **string**: chains of alphanumeric characters flanked by quotation marks, e.g. `'apple'`.   
+You can use either single-quotes or double-quotes, but they must be the same on either end.
+
+Python is able to natively perform certain operations using these datatypes, such as basic addition `+`, subtraction `-`, multiplication `*`, division `/`, etc.
+
+
+```python
+1 + 2 * 3.4 / 5
+```
+
+
+
+
+    2.36
+
+
+
+Python also has some helpful intuitive operations – for example, you can join two strings together with a `+` operator as follows:
+
+
+```python
+'pine' + 'apple' 
+```
+
+
+
+
+    'pineapple'
+
+
+
+Certain operations are not supported by default in Python - for example, you can't add a `str` with an `int`. Running the code below will cause a `TypeError`. Python is quite helpful in explaining errors you might encounter. Read the error message below and see if it makes sense to you.
+
+In general, you should expect to encounter *lots* of these errors, or "bugs", when you start getting into programming. **This is totally normal!**  
+In fact, probably 50% of programming is just squashing bugs.  
+
+If you're ever confused about the source of an error, websites such as [StackOverflow](https://stackoverflow.com/) are great places to look for answers.
```

Comment:
```suggestion
If you're ever confused about the source of an error, websites such as [StackOverflow](https://stackoverflow.com/) are great places to look for answers.
One skill in learning to programming is learning what part of an error message to Google.
```

Or something like this -- indicating that a good step is googling the error message, and that it is a skill to figure out which part is the "right" part to google